plugin/storage/badger/factory.go

```diff
@@ -303,3 +303,9 @@ func (f *Factory) registerBadgerExpvarMetrics(metricsFactory metrics.Factory) {
 		}
 	})
 }
+
+func (f *Factory) CleanUp() error {
```

Comment:
I recommend the name Purge, as it needs to be explicit that all data will be lost ("clean-up" is much weaker).

Also, add a comment explaining that this function is only meant to be called from tests, it's not for production use.